propagators/opentelemetry-propagator-aws-xray/README.md

```diff
@@ -0,0 +1,77 @@
+# OpenTelemetry Propagator AWS X-Ray
+
+[![NPM Published Version][npm-img]][npm-url]
+[![dependencies][dependencies-image]][dependencies-url]
+[![devDependencies][devDependencies-image]][devDependencies-url]
+[![Apache License][license-image]][license-image]
+
+The OpenTelemetry Propagator for AWS X-Ray provides HTTP header propagation for systems that are using AWS `X-Amzn-Trace-Id` format.
+This propagator translates the OpenTelemetry SpanContext into the equivalent AWS header format, for use with the OpenTelemetry JS SDK.
+
+### Installation
+
+```
+npm install --save @opentelemetry/propagator-aws-xray
+```
+
+### Usage
+
+In the [global tracer configuration file](https://github.com/open-telemetry/opentelemetry-js/blob/master/getting-started/README.md#initialize-a-global-tracer), configure the following:
+
+```js
+const { propagation } = require("@opentelemetry/api");
+const { AWSXRayPropagator } = require('@opentelemetry/propagator-aws-xray');
+// ...
+
+module.exports = ("service_name_here") => {
+  // set global propagator
+  propagation.setGlobalPropagator(new AWSXRayPropagator());
+  // ...}
+```
+
+For more details, see the [Getting Started guide](https://aws-otel.github.io/docs/getting-started/javascript-sdk).
+
+### Propagator Details
+
+Example header:`X-Amzn-Trace-Id: Root=1-5759e988-bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=1`
+
+The header consists of three parts: the root trace ID, the parent ID and the sampling decision. The root is required, whereas the parent ID and sampling decision are optional.
+
+#### Root - The AWS X-Ray format trace ID
+
+* Format: (spec-version)-(timestamp)-(UUID)
+    * spec_version - The version of the AWS X-Ray header format. Currently, only "1" is valid.
+    * timestamp - 32-bit number in base16 format, encoded from time(second) when created. Populated by taking the first 8 characters of the OpenTelemetry trace ID.
```

Comment:
Does this mean the `AwsXRayIdGenerator` is mandatory for all monitored entities in case x-ray propagation is used?
I assume yes otherwise the timestamp would be garbage.

Reply:
Nah - for propagation it's just an opaque value so it doesn't need to be a timestamp, if a user doesn't use X-Ray they would still use this propagator with random IDs to propagate through AWS services like SQS.